docs/standard/microservices-architecture/multi-container-microservice-net-applications/implementing-api-gateways-with-ocelot.md

```diff
@@ -0,0 +1,566 @@
+    - ASPNETCORE_ENVIRONMENT=Development
+    - IdentityUrl=http://identity.api              
+  ports:
+    - "5202:80"   
+  volumes:
+    - ./src/ApiGateways/Web.Bff.Shopping/apigw:/app/configuration
+
+webmarketingapigw:
+  environment:
+    - ASPNETCORE_ENVIRONMENT=Development
+    - IdentityUrl=http://identity.api              
+  ports:
+    - "5203:80"   
+  volumes:
+    - ./src/ApiGateways/Web.Bff.Marketing/apigw:/app/configuration
+```
+
+That is why in the source code and shown in the VS Explorer view screenshot below, the only file needed to define each specific business/BFF API Gateway is just a configuration.json file, because the four API Gateways are based on the same Docker image.
+
+![](./media/image34.png)
+
+**Figure 8-33.** The only file needed to define each API Gateway / BFF with Ocelot is a configuration file 
+
+By splitting the API Gateway into multiple API Gateways, different development teams focusing on different subsets of microservices can manage their own API Gateways by using independent Ocelot configuration files while re-using the same Ocelot Docker image. 
+
+Now, if you run eShopOnContainers with the API Gateways (included by default in VS when opening eShopOnContainers-ServicesAndWebApps.sln solution or if running “docker-compose up”), the following sample routes will be performed. 
+
+For instance, when visiting the upstream URL http://localhost:5202/api/v1/c/catalog/items/2/ served by the webshoppingapigw API Gateway, you get the result from the internal Downstream URL http://catalog.api/api/v1/2 within the Docker host, as in the following browser.
+
+![](./media/image35.png)
+
+**Figure 8-34.** Accessing a microservice through a URL provided by the API Gateway 
+
+Because of testing or debugging reasons, if you wanted to directly access to the Catalog Docker container (only at the development environment) without passing through the API Gateway, since “catalog.api” is a DNS resolution internal to the Docker host (service discovery handled by docker-compose service names), the only way to directly access the container is through the external port published in the docker-compose.override.yml, which is provided only for development tests, such as http://localhost:5101/api/v1/Catalog/items/1 in the following browser.
+
+![](./media/image36.png)
+
+**Figure 8-35.** Direct access to a microservice for testing purposes 
+
+But the application is configured so it accesses all the microservices through the API Gateways, not though the direct port “shortcuts”. 
+
+### The Gateway aggregation pattern in eShopOnContainers
+
+As introduced previously, a very flexible way to implement requests aggregation is with custom services, by code. You could also implement request aggregation with the Request Aggregation feature in Ocelot, but it might not be as flexible as you need. Therefore, the selected way to implement aggregation in eShopOnContainers is with an explicit ASP.NET Core Web API services for each aggregator. 
```

Comment:
"an explicit ASP.NET Core Web API services" => either remove "an" or change to "service"